src/coreclr/src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -153,5 +92,67 @@ private static IsolatedComponentLoadContext GetIsolatedComponentLoadContext(stri
 
             return alc;
         }
+
+        private static IntPtr InternalGetFunctionPointer(AssemblyLoadContext alc,
+                                                         string typeName,
+                                                         string methodName,
+                                                         IntPtr delegateTypeNative)
+        {
+            // Create a resolver callback for types.
+            Func<AssemblyName, Assembly> resolver = name => alc.LoadFromAssemblyName(name);
+
+            // Determine the signature of the type. There are 3 possibilities:
+            //  * No delegate type was supplied - use the default (i.e. ComponentEntryPoint).
+            //  * A sentinel value was supplied - the function is marked UnmanagedCallersOnly. This means
+            //      a function pointer can be returned without creating a delegate.
+            //  * A delegate type was supplied - Load the type and create a delegate for that method.
+            Type? delegateType;
+            if (delegateTypeNative == IntPtr.Zero)
+            {
+                delegateType = typeof(ComponentEntryPoint);
+            }
+            else if (delegateTypeNative == (IntPtr)(-1))
+            {
+                delegateType = null;
+            }
+            else
+            {
+                string delegateTypeName = MarshalToString(delegateTypeNative, nameof(delegateTypeNative));
+                delegateType = Type.GetType(delegateTypeName, resolver, null, throwOnError: true)!;
+            }
+
+            // Get the requested type.
+            Type type = Type.GetType(typeName, resolver, null, throwOnError: true)!;
+
+            IntPtr functionPtr;
+            if (delegateType == null)
+            {
+                // Match search semantics of the CreateDelegate() function below.
+                BindingFlags bindingFlags = BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic;
+                MethodInfo? methodInfo = type.GetMethod(methodName, bindingFlags);
+                if (methodInfo == null)
+                    throw new MissingMethodException(typeName, methodName);
+
+                // Verify the function is properly marked.
+                if (null == methodInfo.GetCustomAttribute<UnmanagedCallersOnlyAttribute>())
```

Comment:
I think this can be changed to `methodInfo.IsDefined(typeof(UnmanagedCallersOnlyAttribute), false)`. If memory serves @jkotas indicated this was a cheaper call at one point and when I originally wrote this I didn't know that.

Reply:
I don't want to make this change in this PR. The equivalent code is `methodInfo.IsDefined(typeof(UnmanagedCallersOnlyAttribute), true)`.